app/articles/server/methods/admin.js

```diff
@@ -1,17 +1,77 @@
 import { Meteor } from 'meteor/meteor';
+import { HTTP } from 'meteor/http';
 
 import { settings } from '../../../settings';
 
+const blogUrl = 'http://localhost:2368/ghost/api/v2/admin/authentication/setup/';
+const sessionUrl = 'http://localhost:2368/ghost/api/v2/admin/session/';
+
 Meteor.methods({
 	Articles_admin_panel() {
 		const enabled = settings.get('Articles_enabled');
 
 		if (!enabled) {
 			throw new Meteor.Error('Articles are disabled');
 		}
-
-		return {
-			message: 'Redirecting to articles admin panel',
+		const user = Meteor.users.findOne(Meteor.userId());
+		const options = {
+			data: {
+				username: user.emails[0].address,
+				password: 'qwe123qwe123',
+			},
+			headers: {
+				'Content-Type': 'application/json',
+				Referer: 'http://localhost:2368/ghost/',
+			},
 		};
+		let response = HTTP.call('GET', blogUrl);
+		if (response.data && response.data.setup && response.data.setup[0]) {
```

Comment:
We can simplify the the if block via using try/catch to evaluate first setup item. After that for **Ghost already setup** & **initial Ghost setup** scenarios we can handle by splitting function.
e.g:
```
let setupFirstItem;
try {
    setupFirstItem = response.data.setup[0];
} catch(e) {
    throw new Meteor.Error('Unable to redirect. Make sure Ghost is running');
}
if (setupFirstItem) {
    // Ghost site is already setup
} else {
   // Setup Ghost Site and set title
}
```